controllers/setupcontroller/setup_controller.go

```diff
@@ -0,0 +1,83 @@
+package setupcontroller
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
+	odhClient "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/client"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/upgrade"
+)
+
+type SetupControllerReconciler struct {
+	*odhClient.Client
+}
+
+func (r *SetupControllerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := logf.FromContext(ctx).WithName("SetupController")
+	log.Info("Reconciling setup controller", "Request.Name", req.Name) // log.V(1).Info(...)
```

Comment:
`"Request.Name", req.Name` is probably redundant as it should be included already in the log passed to the within the context. Can you confirm ?

Reply:
Yeah you're correct it does get passed from within the context. 
This is the log
```
{"level":"info","ts":"2024-12-16T14:00:12Z","logger":"SetupContoller","msg":"Reconciling setup controller","controller":"configmap","controllerGroup":"","controllerKind":"ConfigMap","ConfigMap":{"name":"delete-configmap","namespace":"opendatahub-operator-system"},"namespace":"opendatahub-operator-system","name":"delete-configmap","reconcileID":"1c11083d-f086-4198-bfde-ba26a7f7fc38","Request.Name":"delete-configmap"}
```

Reply:
This is similar to that in DSC controller too. But maybe it makes sense for the DSC and not for the setup controller? Or should it be removed from there too.

Reply:
Probably it should be revisited in the DSC controller too, for the moment I would just fix it here

Reply:
Done 👍